python/cugraph/cugraph/dask/common/input_utils.py

```diff
@@ -102,8 +104,20 @@ def create(cls, data, client=None, batch_enabled=False):
         else:
             raise TypeError("Graph data must be dask-cudf dataframe")
 
+        broadcast_worker = None
+        if batch_enabled:
+            worker_ranks = client.run(_get_nvml_device_index)
```

Comment:
I think this will fail on MNMG setups. The `worker_rank`  here gets `nvml_device_index` which will be zero for 2 devices (the first device on both nodes) . 

I think we should figure out a way to get this from raft instead.

Reply:
How about we directly fetch from the upstream `raft_dask` library to prevent breaking in the future too ? 
```suggestion
            from raft_dask.common.comms import _func_worker_ranks
            worker_ranks = _func_worker_ranks(client)
```

Reply:
That PR is not merged yet. That is why I added a FIXME in line 286

Reply:
> I think this will fail on MNMG setups. The worker_rank here gets nvml_device_index which will be zero for 2 devices (the first device on both nodes)

Why would it be zero? And I extensively tested this PR on 2 and 4 nodes and all runs passed

Reply:
> That PR is not merged yet. That is why I added a FIXME in line 286
No need to wait for your PR to be merged because `_func_worker_ranks` already exits. Its implementation will just change once your PR is merge.

> I think this will fail on MNMG setups
I don't think this can fail because the address of rank 0 in this PR `always` always matches the raft one. In fact, `client.run(_get_nvml_device_index)` is deterministic and always returns the result in increasing network ID or IP address number(the keys of the dictionary). On a MNMG run, even though there are multiple `rank 0`, this PR will always pick the one in the first node which is consistent with the raft one. This is true because the raft PR only applies the rank offsets to the second node and above. Furthermore, I extensively tested this PR on Friday on both 2 and 4 nodes and all runs passed.


> How about we directly fetch from the upstream raft_dask library to prevent breaking in the future too ?
Right and this is what I had in mind when I added a FIXME. This will avoid code duplication and prevent breaking changes.

Reply:
> On a MNMG run, even though there are multiple pairs with rank 0, this PR will always pick the one in the first node which is consistent with the raft one
Does this always hold true, Maybe i am missing the logic on this

Reply:
As we discussed, getting the ranks that were established after the comms initialization guarantee consistency.